### PR TITLE
Add MessageAdapter content parts serialization test coverage

### PR DESCRIPTION
**Pull request by smola**
# What Does This Do

# Motivation

# Additional Notes

# Contributor Checklist

- Format the title according to [the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any other useful labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- Avoid using `close`, `fix`, or [any linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- Update the [CODEOWNERS](https://github.com/DataDog/dd-trace-java/blob/master/.github/CODEOWNERS) file on source file addition, migration, or deletion
- Update [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) with any new configuration flags or behaviors
- Add your completed PR to the merge queue by commenting `/merge`. You can also:
  - Customize the commit message associated with the merge with `/merge --commit-message "..."`
  - Remove your PR from the merge queue with `/merge -c`
  - Skip all merge queue checks with `/merge -f --reason "reason"`; please use this judiciously, as some checks do not run at the PR-level (note: the PR still needs to be mergeable, this will only skip the pre-merge build)
  - Get more information in [this doc](https://datadoghq.atlassian.net/wiki/spaces/DEVX/pages/3121612126/MergeQueue)

Jira ticket: [PROJ-IDENT]

### CODE REVIEW

**Comment by smola** [comment]
@codex review

File: dd-java-agent/agent-aiguard/src/test/groovy/com/datadog/aiguard/AIGuardInternalTests.groovy [hunk 1]
```diff
@@ -861,6 +861,25 @@ class AIGuardInternalTests extends DDSpecification {
     }
   }
```
Comment (jandro996) [inline dd-java-agent/agent-aiguard/src/test/groovy/com/datadog/aiguard/AIGuardInternalTests.groovy:863]: nit: the adapter is re-created on every `where:` row. Consider moving it to a `@Shared` field.

**Review by jandro996 — APPROVED** [review]
LGTM! just some nits

File: dd-java-agent/agent-aiguard/src/test/groovy/com/datadog/aiguard/AIGuardInternalTests.groovy [hunk 2]
```diff
@@ -861,6 +861,25 @@ class AIGuardInternalTests extends DDSpecification {
...
+
+    expect:
+    JSONAssert.assertEquals(expected, adapter.toJson(message), JSONCompareMode.NON_EXTENSIBLE)
+
+    where:
+    message                                                                            | expected
+    AIGuard.Message.message('user', [AIGuard.ContentPart.text('Hello world')])         | '{"role": "user", "content": [{"type": "text", "text": "Hello world"}]}'
+    AIGuard.Message.message('user', [AIGuard.ContentPart.imageUrl('https://example.com/image.jpg')]) | '{"role": "user", "content": [{"type": "image_url", "image_url": {"url": "https://example.com/image.jpg"}}]}'
+    AIGuard.Message.message('user', [
+      AIGuard.ContentPart.text('Describe this image:'),
```
Comment (jandro996) [inline dd-java-agent/agent-aiguard/src/test/groovy/com/datadog/aiguard/AIGuardInternalTests.groovy:880]: nit: missing edge case - `AIGuard.Message.message('user', [])` is valid (constructor allows empty lists) and would serialize to `"content": []`. Trivial to add as an extra row.

File: dd-java-agent/agent-aiguard/src/test/groovy/com/datadog/aiguard/AIGuardInternalTests.groovy [hunk 3]
```diff
@@ -861,6 +861,25 @@ class AIGuardInternalTests extends DDSpecification {
     }
   }
 
+  void 'test adapter serializes content parts'() {
```
[the remark this answers is not in the document]
Comment (smola) [reply dd-java-agent/agent-aiguard/src/test/groovy/com/datadog/aiguard/AIGuardInternalTests.groovy:864]: No. Not migrating it in this PR.

File: dd-java-agent/agent-aiguard/src/test/groovy/com/datadog/aiguard/AIGuardInternalTests.groovy (hunk 2 above)
Comment (smola) [reply dd-java-agent/agent-aiguard/src/test/groovy/com/datadog/aiguard/AIGuardInternalTests.groovy:880]: Thanks, done.

**Comment by smola** [comment]
@codex review

File: dd-java-agent/agent-aiguard/src/test/groovy/com/datadog/aiguard/AIGuardInternalTests.groovy [hunk 4]
```diff
@@ -861,6 +861,28 @@ class AIGuardInternalTests extends DDSpecification {
...
```
[the remark this answers is not in the document]
Comment (smola) [reply dd-java-agent/agent-aiguard/src/test/groovy/com/datadog/aiguard/AIGuardInternalTests.groovy:864]: Nope. Nope. Nope.

[the remark this answers is not in the document]
Comment (bric3) [reply dd-java-agent/agent-aiguard/src/test/groovy/com/datadog/aiguard/AIGuardInternalTests.groovy:864]: In a way codex is right :)
